api/client/trust.go

```diff
@@ -375,34 +375,52 @@ func (cli *DockerCli) trustedPush(repoInfo *registry.RepositoryInfo, tag string,
 
 	defer responseBody.Close()
 
-	targets := []target{}
+	// If it is a trusted push we would like to find the target entry which match the
+	// tag provided in the function and then do an AddTarget later.
+	target := &client.Target{}
+	// Count the times of calling for handleTarget,
+	// if it is called more that once, that should be considered an error in a trusted push.
+	cnt := 0
 	handleTarget := func(aux *json.RawMessage) {
+		cnt++
```

Comment:
@endophage actually it will not have a chance to execute more than once since it will return when `tag` is empty.

This `cnt` check is base on https://github.com/docker/docker/pull/20307#discussion_r53051231